src/Error/Middleware/ErrorHandlerMiddleware.php

```diff
@@ -201,4 +224,20 @@ protected function getErrorHandler(): ErrorHandler
 
         return $this->errorHandler;
     }
+
+    /**
+     * Get a exception trap instance
+     *
+     * @return \Cake\Error\ExceptionTrap The exception trap.
+     */
+    protected function getExceptionTrap(): ExceptionTrap
+    {
+        if ($this->exceptionTrap === null) {
+            /** @var class-string<\Cake\Error\ExceptionTrap> $className */
+            $className = App::className('ExceptionTrap', 'Error');
+            $this->exceptionTrap = new $className($this->getConfig());
```

Comment:
So, won't calling this override the non-trap configured in the constructor?

Reply:
It would. I'm relying on this method not being called when there is an `errorHandler` set. I could make that more explicit in this method by raising an error if the errorhandler is defined. What do you think?

Reply:
It's an edge case that most people won't encounter, but since its a public getter, we should probably error.